Added support for task arguments

stevenvachon opened the pull request:
This is waiting on #46

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/m-zuber/npm-watch/43)
<!-- Reviewable:end -->

stevenvachon committed:
Rewrite in ES2015/ES6

…with a more reusable API that can facilitate tests.

M-Zuber commented:
Overall this looks good, but I found the following edge cases that would still need to work:
- `npm run watch` should simply run all tasks defined
- `npm run watch -- ...args` should run all tasks defined passing in the args to each one

here is the setup I was using to test it
```json

stevenvachon commented:
We should add a test suite.

stevenvachon committed:
Remove package-lock.json

fixes #44

M-Zuber commented:
> We should add a test suite.

That is something I would love to do, if you have an idea on how to implement, please open an issue/pr so we can discuss it in a dedicated location

stevenvachon commented:
I'd like to do it here in a separate commit, as it'll help me be sure that my PR passes your above case.

M-Zuber commented:
Generally I would prefer to separate something like that out to a dedicated PR, but as long as each commit is self contained, go ahead and do it on this PR

stevenvachon committed:
Add support for task arguments

…when task name is specified. It is not currently possible to support arguments without a task because argv[2] was hard-coded to be the task name. If it were defined as “--arg”, it would treat that as the task name.

fixes #37

stevenvachon committed:
Add test suite

fixes #42,#45

stevenvachon committed:
Reduce published footprint

…by only exporting what is necessary for production use.

M-Zuber commented:
Travis and AV are both connected now
